hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java

```diff
@@ -1254,8 +1255,10 @@ private boolean processAllResponses(final Connection connection) throws IOExcept
           if (call == null) {
             return true;
           }
+          metrics.removeCallFromResponseQueue(call.response.getRemaining());
```

Comment:
Why add in multiple places, looks like processResponse(call) is the place where the actual change is happening, so move most updates to that method?

  if (!call.response.hasRemaining()) {
        call.done();
       // Remove here <====
        return true;
      } else {
       // update here <=====
        return false; // Socket can't take more, we will have to come back.
      }

Reply:
If the response queue is empty at the beginning, call will be directly processed. After processed, the metrics will become -1.

Reply:
```
          if (call.connection.responseQueue.isEmpty()) {
            // If we're alone, we can try to do a direct call to the socket. It's
            //  an optimisation to save on context switches and data transfer between cores..
            if (processResponse(call)) {
              return; // we're done.
            }
            // ..
          }
```